server/data/prisonRegisterStore.ts

```diff
@@ -0,0 +1,31 @@
+import { RedisClient } from './redisClient'
+import { PrisonDto } from './interfaces/prisonRegisterApi/prisonRegisterApiTypes'
+import logger from '../../logger'
+
+const ACTIVE_PRISONS = 'activePrisons'
+
+export default class PrisonRegisterStore {
+  constructor(private readonly client: RedisClient) {
+    client.on('error', error => {
+      logger.error(error, `Redis error`)
+    })
+  }
+
+  private async ensureConnected() {
+    if (!this.client.isOpen) {
+      await this.client.connect()
+    }
+  }
+
+  async setActivePrisons(prisons: Array<PrisonDto>, durationDays = 1): Promise<string> {
+    await this.ensureConnected()
+    const activePrisons = prisons.filter(prison => prison.active === true)
```

Comment:
Extremely minor, but wondering if this filter logic belongs in this set method? I think I might expect this method to simply set whatever prisons I passed it as the "active prisons".

Reply:
Yes, I think you are right ....